build_src.cmake

```diff
@@ -166,9 +166,17 @@ set(LIBXMP_SRC_LIST_DEPACKERS
     src/depackers/xfnmatch.c
     src/depackers/xz_dec_lzma2.c
     src/depackers/xz_dec_stream.c
-# AMIGA
-    src/depackers/xfd.c
-    src/depackers/xfd_link.c
-# WIN32
-    src/depackers/ptpopen.c
 )
+
+if(WIN32)
+    list(APPEND LIBXMP_SRC_LIST_DEPACKERS
+        src/depackers/ptpopen.c
```

Comment:
What I mean is: apart from `win32.c`, none of other amiga or windows sources should not be needed.

Reply:
OK, correction: Please revert this: you are at the _wrong_ file.

My criticism was for test-dev/CMakefile.txt !

Reply:
ptpopen is required:
![изображение](https://user-images.githubusercontent.com/6751442/127392734-b14bc717-a573-4ddb-802f-59f8c6219cc0.png)

depacker.c depends on ptpopen and this will fail if I don't include this.

Also:
![изображение](https://user-images.githubusercontent.com/6751442/127393259-4efd318b-cbd2-44c8-847c-255cf1a16a72.png)

Reply:
And then:
![изображение](https://user-images.githubusercontent.com/6751442/127393514-7753e82c-e91b-43d6-b0b7-865f42c99305.png)

Reply:
And also:
![изображение](https://user-images.githubusercontent.com/6751442/127393628-e476a360-4445-44ea-8db7-a1f322385256.png)

So, these files were getting built. Makefiles were requiring these files inexplicitly, and so, at CMake, I should require these files explicitly.

Reply:
> ptpopen is required:

No, it is not: it is required in libxmp library build, _not_ in test-dev/libxmp-tests.exe

> depacker.c depends on ptpopen and this will fail if I don't include this.

See above

Reply:
The same against Amiga files:
![изображение](https://user-images.githubusercontent.com/6751442/127393818-86c512de-7f4b-4df6-9c25-98828afe5ba0.png)
they were appear in the build process and they all included inexlicitly by Makefiles

Reply:
Okay, I reverted back and made the fix at the tests-dev

Reply:
As I said above: you are talking about the _wrong_ thing!

Revert https://github.com/libxmp/libxmp/commit/82c4e0d5f1a0f891b10f08c4e1130b77c0a5de56, and then apply the following: you'll see that it will not fail:
```diff
diff --git a/test-dev/CMakeLists.txt b/test-dev/CMakeLists.txt
index 4dedec8..70790d0 100644
--- a/test-dev/CMakeLists.txt
+++ b/test-dev/CMakeLists.txt
@@ -27,12 +27,7 @@ set(DEVTEST_SRC
     ../src/loaders/sample.c
     ../src/loaders/common.c
     ../src/depackers/xfnmatch.c
-# AMIGA
-    ../src/depackers/xfd.c
-    ../src/depackers/xfd_link.c
-# WIN32 OR WATCOM
     ../src/win32.c
-    ../src/depackers/ptpopen.c
 )
 
 set(DEV_GENERATED "${CMAKE_CURRENT_BINARY_DIR}/tests-dev-generated")
```

Reply:
Yes, that's the correct solution.

Reply:
> As I said above: you are talking about the wrong thing!

I received the email notification later than I write my answer at your first notice